testdata/scripts/scriptsubdir_unix.txt

```diff
@@ -14,10 +14,13 @@ exec tar -tzf archive.tar.gz
 cmp stdout golden/archive
 
 -- golden/apply --
+$HOME
 $HOME/dir
 -- golden/archive --
+otherdir/script
 dir/
 dir/script
+otherdir/
```

Comment:
I'm a bit surprised that this test passes as chezmoi normally sorts filenames alphabetically to give repeatable output.

Reply:
Yeah, I'm glad you mentioned that, I was surprised too.  I eventually convinced myself it was `tar` that wasn't sorting them, and that's why it was in this order.  But I could be wrong about that.

Reply:
Actually, I just realized that what happens is that chezmoi emits the diff in the order that the changes are applied, which is normally in alphabetical order. However, this is a `before_` script, so it's run before the destination directory is updated, which is why `otherdir/script` appears first.